commit d97a80a73b85080e04e78f9a66a594cf4b550bac
Author: Qijia Liu <liumeo@pku.edu.cn>
Date:   Fri Jan 2 22:19:46 2026 -0500

    log and use U+FFFD for invalid utf-8 in Text::append

diff --git a/src/lib/fcitx/text.cpp b/src/lib/fcitx/text.cpp
index 852516ff..78a9493c 100644
--- a/src/lib/fcitx/text.cpp
+++ b/src/lib/fcitx/text.cpp
@@ -7,9 +7,9 @@
 
 #include "text.h"
 #include <iterator>
-#include <stdexcept>
 #include <tuple>
 #include <vector>
+#include "fcitx-utils/log.h"
 #include "fcitx-utils/stringutils.h"
 #include "fcitx-utils/utf8.h"
 
@@ -51,7 +51,9 @@ void Text::setCursor(int pos) {
 void Text::append(std::string str, TextFormatFlags flag) {
     FCITX_D();
     if (!utf8::validate(str)) {
-        throw std::invalid_argument("Invalid utf8 string");
+        FCITX_ERROR() << "Invalid utf8 string: " << str;
+        d->texts_.emplace_back("ï¿½", flag);
+        return;
     }
     d->texts_.emplace_back(std::move(str), flag);
 }
